use_cases/fetcher.go

```diff
@@ -19,26 +21,65 @@ type Fetcher struct {
 	storage writer
 }
 
+type result struct {
+	Error   error
+	Pokemon models.Pokemon
+}
+
 func NewFetcher(api api, storage writer) Fetcher {
 	return Fetcher{api, storage}
 }
 
 func (f Fetcher) Fetch(from, to int) error {
 	var pokemons []models.Pokemon
-	for id := from; id <= to; id++ {
-		pokemon, err := f.api.FetchPokemon(id)
-		if err != nil {
-			return err
-		}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
Why are you deferring the cancel method? cancel the context should happen on error not just because of deferring.